aperoll/star_field_items.py

```diff
@@ -0,0 +1,233 @@
+        )
+        gui_stars = cat[(cat["type"] == "GUI") | (cat["type"] == "BOT")]
+        acq_stars = cat[(cat["type"] == "ACQ") | (cat["type"] == "BOT")]
+        fids = cat[cat["type"] == "FID"]
+        mon_wins = cat[cat["type"] == "MON"]
+
+        self.star_labels = [StarcatLabel(star, self) for star in cat]
+        self.guide_stars = [GuideStar(gui_star, self) for gui_star in gui_stars]
+        self.acq_stars = [AcqStar(acq_star, self) for acq_star in acq_stars]
+        self.mon_stars = [MonBox(mon_box, self) for mon_box in mon_wins]
+        self.fid_lights = [FidLight(fid, self) for fid in fids]
+
+    def setPos(self, *_args, **_kwargs):
+        # the position of the catalog is ALLWAYS (0,0)
+        pass
+
+    def set_pos_for_attitude(self, attitude):
+        """
+        Set the position of all items in the catalog for a given attitude.
+
+        Calling QGraphicsItem.set_pos would not work. Children positions are relative to the
+        parent, but in reality the relative distances between items changes with the attitude.
+        One cannot change the position of a single item and then get the rest as a relative shift.
+        Each item needs to be set individually.
+        """
+        # item positions are relative to the parent's position (self)
+        # but the parent's position is (or should be) always (0, 0)
+        for item in self.childItems():
+            yag, zag = radec_to_yagzag(
+                item.starcat_row["ra"], item.starcat_row["dec"], attitude
+            )
+            row, col = yagzag_to_pixels(yag, zag, allow_bad=True)
+            # item.setPos(-yag, -zag)
+            item.setPos(row, -col)
+
+    def boundingRect(self):
+        return QtC.QRectF(0, 0, 1, 1)
+
+    def paint(self, _painter, _option, _widget):
+        # this item draws nothing, it just holds children
+        pass
+
+    def __repr__(self):
+        return repr(self.starcat)
+
+
+class FidLight(QtW.QGraphicsEllipseItem):
+    def __init__(self, fid, parent=None):
+        self.starcat_row = fid
+        s = 27
+        w = 3
+        rect = QtC.QRectF(-s, -s, 2 * s, 2 * s)
+        super().__init__(rect, parent)
+        self.fid = fid
+        pen = QtG.QPen(QtG.QColor("red"), w)
+        self.setPen(pen)
+        # self.setPos(-fid["yang"], -fid["zang"])
+        self.setPos(fid["row"], -fid["col"])
+
+        line = QtW.QGraphicsLineItem(-s, 0, s, 0, self)
+        line.setPen(pen)
+        line = QtW.QGraphicsLineItem(0, -s, 0, s, self)
+        line.setPen(pen)
+
+
+class StarcatLabel(QtW.QGraphicsTextItem):
+    def __init__(self, star, parent=None):
+        self.starcat_row = star
+        super().__init__(f"{star['idx']}", parent)
+        self._offset = 30
+        self.setFont(QtG.QFont("Arial", 30))
+        self.setDefaultTextColor(QtG.QColor("red"))
+        # self.setPos(-star["yang"], -star["zang"])
+        self.setPos(star["row"], -star["col"])
+
+    def setPos(self, x, y):
```

Comment:
And if everything basically gets plugged in as (row, -col) - would there be value in flipping the function signature and adding a docstring?

Reply:
`setPos` is not written by me. It is the Qt function, and I don't want to start overriding all `setPos` functions for all possible items, and we do want to keep the appearance, so I do not see a way other than having minus signs all over.

There might be some kind of hack, but I prefer to keep it explicit.

Reply:
I mean `setPos` for all other `QGraphicsItem`s. This one is overridden but it is an exception.

Reply:
Okey Doke.